### PR TITLE
Enable caldav push notifications by default

### LINKED ISSUE #39862: Enable caldav push notifications by default
### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are interested into the same feature.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments. 

**Is your feature request related to a problem? Please describe.**

As a user of Nextcloud who doesn't use connected clients I want to see a web notification for calendar events.

**Describe the solution you'd like**

Revert https://github.com/nextcloud/server/pull/16970

**Describe alternatives you've considered**

https://github.com/nextcloud/server/issues/39158

**Additional context**

Needs admin upgrade docs because it will have a big influence on existing setups.

### PR DESCRIPTION
**Pull request by hamza221**
* Resolves: [#39862](https://github.com/nextcloud/server/issues/39862)

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [ ] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [ ] Screenshots before/after for front-end changes
- [ ] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [ ] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

### CODE REVIEW

**Comment by hamza221** [comment]
> I don't understand how it answers #39158, sorry. :(

it doesn't, it answers https://github.com/nextcloud/server/issues/39862

**Review by GretaD — APPROVED** [review]
i only checked the vue and js changes

**Review by miaulalala — APPROVED** [review]

File: apps/dav/lib/CalDAV/Reminder/NotificationProvider/PushProvider.php [hunk 1]
```diff
@@ -82,10 +82,6 @@ public function send(VEvent $vevent,
 						 ?string $calendarDisplayName,
 						 array $principalEmailAddresses,
 						 array $users = []):void {
-		if ($this->config->getAppValue('dav', 'sendEventRemindersPush', 'no') !== 'yes') {
```
Comment (ChristophWurst) [inline apps/dav/lib/CalDAV/Reminder/NotificationProvider/PushProvider.php:85]: like @tcitworld said we should flip the default value but leave the option to turn the feature *off*

**Review by ChristophWurst — CHANGES_REQUESTED** [review]

**Review by tcitworld — APPROVED** [review]

**Review by ChristophWurst — APPROVED** [review]

**Comment by ChristophWurst** [comment]
Failure of ``OCA\DAV\Tests\unit\CalDAV\Reminder\NotificationProvider\PushProviderTest::testNotSend`` and similar are related

**Review by ChristophWurst — APPROVED** [review]
Time to squash

**Comment by ChristophWurst** [comment]
Needs admin docs for how to set sendEventRemindersPush and an entry in the admin upgrade section for 28

**Pull request merged**